src/main/js/utils/generate.js

```diff
@@ -14,50 +15,55 @@ const generateSchedules = (values, loadQuery) => (dispatch, getState) => {
   const { courseList } = getState().courses;
   if (courseList.length !== 0) {
     dispatch(startGenerating());
-    const filteredResults = filterCourses(values, courseList);
-    dispatch(filteredCourses(filteredResults));
-    const filteredCourseList = filteredResults.map(result => (
-      result.filtered));
-    let schedules = [];
-    if (!filteredCourseList.find(list => list.length === 0)) {
-      const scheduleMaker = new ScheduleMaker(filteredCourseList, parseInt(values.gap, 10));
-      try {
-        schedules = scheduleMaker.makeSchedules();
-      } catch (e) {
-        schedules = e;
+    addGPA(courseList).then((res) => {
```

Comment:
Wouldn't it be more efficient to filter the courseList first? Also we only need to addGPA when the value is toggled right?